### PR TITLE
iss30: add mobx BalanceService combined with chart

### PR DESCRIPTION
**Pull request by lukaswski**

### CODE REVIEW

File: src/components/BalanceService.tsx [hunk 1]
```diff
@@ -14,9 +15,11 @@ class BalanceService {
     constructor(memoryService: MemoryService) {
         this.memoryService = MemoryService.instance
         this.balances = [];
+        makeAutoObservable(this)
     };
 
     static instance = new BalanceService(MemoryService.instance);
 };
+export const myBalance = new BalanceService()
```
Comment (oneacik) [inline src/components/BalanceService.tsx]: instead of exporting myBalance, use BalanceService.instance.

File: src/App.tsx [hunk 2]
```diff
@@ -3,17 +3,20 @@ import Chart from "./components/Chart";
...
-const App = () => (
-    <div className="App">
+
+const App = observer(({ balances }) => (
+  <div className="App">
+    {console.log(balances)}
         <StatedBalanceForm balanceService={BalanceService.instance}/>
         <Chart
-            values={[[new Date("November 17, 2020 03:24:00"), 20.20], [new Date("November 17, 2020 06:24:00"), 20.20], [new Date("November 19, 2020 12:24:00"), 42.20], [new Date("November 20, 2020 12:24:00"), 53.20]]}/>
+            values={balances.memoryService.getValues().map((data) => Object.values(data))}/>
```
Comment (oneacik) [inline src/App.tsx]: memoryService.getValues is not going to work, as it is reading data from external storage and MobX can't track it. Change it to balances.getValues.

Btw. why Object.values here?

File: src/components/BalanceService.tsx [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import MemoryService from "../core/Memory";
+import { makeAutoObservable } from "mobx"
 
 class BalanceService {
     balances: { date: Date, value: number }[];
```
Comment (oneacik) [inline src/components/BalanceService.tsx:5]: It should contain method getBalances() that should be used to access balances variable.
Then mobx can track its changes.

File: src/App.tsx [hunk 4]
```diff
@@ -3,17 +3,20 @@ import Chart from "./components/Chart";
...
 import { BrowserRouter, Route, Switch } from "react-router-dom";
 
 import Navigation from "./components/pages/Navigation";
 import NameForm from "./components/pages/StatedNameForm";
 import GraphForm from "./components/pages/StatedGraphForm";
 
-const App = () => (
-    <div className="App">
+
+const App = observer(({ balances }) => (
```
Comment (oneacik) [inline src/App.tsx]: just pls rename it to balanceService.
App looks like nice factory pattern.

File: src/App.tsx (hunk 2 above)
Comment (oneacik) [reply src/App.tsx]: btw. in future there will be StatedChart that should be supplied with whole memoryService + StatedChart should be observer too.

File: src/components/BalanceService.tsx (hunk 1 above)
Comment (lukaswski) [reply src/components/BalanceService.tsx]: ok
done

File: src/App.tsx (hunk 4 above)
Comment (lukaswski) [reply src/App.tsx]: ok done

**Pull request closed without merging**